### DISCUSSION TITLE
How to perform the flexible reanalysis?

### DISCUSSION THREAD

**Discussion by MatteoLacki** [vdemichev/DiaNN#178] [General]
Hello!

Under `Output/Flexible reanalysis` section of the manual it is mentioned that the computations can be stopped after the computationally-demanding part of processing separately each run.

I would like to analyze with DiaNN one measurement a time and produce the necessary .quant files separately for each run (as it might take some time for the data  to be collected, or the runs might be acquired in parallel on more than one different instruments).

What is the command I have have to use with `diann-1.8` on linux in the terminal to produce such a file and only these? And what would be the command to later on, when all data is there and all .quant files are produced (I can check that indpenendently) to analyze the data together with the MBR algorithm and all other goodies on top?

Best wishes,

Matteo Lacki

**Comment by vdemichev** [comment]
Hi Matteo,

DIA-NN always produces .quant files automatically, no special command needed.
The --use-quant option, to reuse the .quant files, is equivalent to the respective GUI checkbox.
MBR will reanalyse the whole dataset. The .quant files from the first pass will be reused, and when using MBR they are only saved for the first pass.

In general, I recommend copying the command line generated by the GUI, to figure out which commands to use. You can run the GUI for thus purpose both on Windows or Linux (under Wine 6.8 or later).

It might also be useful to check the info on incremental data processing here: https://github.com/vdemichev/DiaNN#frequently-asked-questions

Best,
Vadim

**Reply by MatteoLacki** [reply]
Hello again,

I have read this, but I need to make sure that I get it right.
When I open GUI there is no just-do-what-I-want button :)
Also using GUI I have to know, what to turn off, and what needs to remain the same.
At least, I don't know it yet that much, so I will still ask some small questions here, as other silly people can get into the same place I am :)

1. OK, I run whatever command and get the .quant file.
I get it right, that I can then simply move these into one folder for reanalysis and this folder can be specified after the `--temp` flag, right?

2. While reanalysing, I use the `--reanalyse` flag for MBR and `--use-quant` to actually reuse the .quant files, right? Also, while running DiaNN for the first time, these flags should not be included, simply because I cannot have MBR on a single measurement and I definitely don't have any quant files yet, because the data was just acquired.

3. To do MBR with some specific flags set, e.g. `--min-fr-mz 200` the same setting should be used in the initial .quant-file-producing run, right? The same for most other parameters.
What about `--no-maxlfq` here? Can I run first runs with this flag and then MBR without it?

I am sorry for getting into such details, but they are really important to us.
What can I say, I can only promise that all of our papers will definitely quote DiaNN papers, and since we are automating stuff, there should be more papers than if we wouldn't try that :)

Very best wishes,

**Reply by vdemichev** [reply]
1. Yes.
2. Yes.
3. --min-fr-mz controls precursor generation based on a FASTA database - it must be set to a particular value from the very start (or left default), cannot be changed when reusing the .quant files. Yes, --no-maxlfq only affects the very last cross-run step, it has no effect while generating .quant files.

No problem, please don't hesitate to post any questions here.

Best,
Vadim

**Comment by MalteSielaff** [comment]
Hi Vadim,

I have a question that goes in a very similar direction, so I'll use this conversation to ask it :)

I processed a more or less large diaPASEF-data set in several batches. Now I would like to perform a global analysis across all runs and my idea was to use the .quant files to speed up the process. However, I realized that I might have selected the "wrong" **protein inference** and **quantification strategy** parameter for this dataset during the batch-wise processing. In addition, I initially used **--relaxed-prot-inf** and would like to repeat the analysis without that option for comparison.
Do these parameters (protein inference, quantification strategy, --relaxed-prot-inf) affect the generation of the .quant files or can I safely change them in the combined diaNN analysis using the existing .quant files?

Many thanks,
Malte

**Reply by vdemichev** [reply]
Hi Malte,

Protein inference or --relaxed-prot-inf do not affect .quant files. But quantification strategy does.

Best,
Vadim
